# Read in a FM produced by gcfish and output errors for the plot of surveys

We will import a Fishmat reading function below. The reading function will expect an informative header in the fm file.

In [1]:
# Imports and function definition
import numpy as np
import scipy.linalg
import gcfishlib as gfl

# Options
marged = True
fishfile = 'in/GC-Fish_linear-248468740_8ea09e8_experiment_photz_WISE_yellow_1perc.txt'; da = 'Da'

# Retrieve the Fisher matrix, parameters, indices, and redshifts
fm, parameters, indices, zs = gfl.read_fisher(fishfile, marged)

In [2]:
# Print the fracional errors on DV
errors, zs = gfl.DV_from_fm(fm, parameters, indices, zs, da, marged)
print('z =', ', '.join([str(e) for e in gfl.roundsf(zs,3)]))
print('sigDV/DV =', ', '.join([str(e) for e in gfl.roundsf(errors,3)]))

z = 0.28, 0.66
sigDV/DV = 0.018, 0.0163


In [4]:
# Print the fracional errors on DA
errors, zs = gfl.DA_from_fm(fm, parameters, indices, zs, da, marged)
print('z =', ', '.join([str(e) for e in gfl.roundsf(zs,3)]))
print('sigDA/DA =', ', '.join([str(e) for e in gfl.roundsf(errors,3)]))

z = 0.28, 0.66
sigDA/DA = 0.00774, 0.00658


For reference:  

DESI LRGs (z=0.51) if I randomly take redshift error as 0.0001?  
sigDV/DV = 0.00658  
sigDA/DA = 0.00859  
And with z-error at 0.001
sigDV/DV = 0.00717  
sigDA/DA = 0.00919  
But the correct dz for DESI is about 0.0005 according to the DESI "redbook"?, which gives:  

 

WISE with 3 bins and dz=0.1 (fixing shot noise)

sigDV/DV = 0.122, 0.146, 0.742  
sigDA/DA = 0.0249, 0.0216, 0.0901  

Euclid with 13 bins and dz=0.001 (fixing shot noise)

sigDV/DV = 0.00554, 0.00553, 0.00557, 0.00565, 0.00592, 0.00637, 0.00699, 0.0078, 0.00934  
sigDA/DA = 0.00447, 0.00449, 0.00455, 0.00464, 0.0049, 0.00529, 0.00584, 0.00654, 0.00786  


Final C2-NL2 results from Domenico's file:

`sigDV/DV = 0.00942, 0.00917, 0.00903, 0.00897, 0.00924, 0.00977, 0.0106, 0.0117, 0.0139`

Dida Markovic, 2017 -> 2025